src/routeGeneration/templateHelpers.ts

```diff
@@ -359,7 +361,16 @@ export class ValidationService {
     return new Buffer(value);
   }
 
-  public validateModel(name: string, value: any, refName: string, fieldErrors: FieldErrors, parent = ''): any {
+  public validateModel( input: {
```

Comment:
I was getting all kinds of problems when trying to add a required property after an optional property, so I switched over to an object input so that the parameters don't have an order. This also helped with readability in the tests. I think this approach is sometimes called "named properties."

Reply:
minimalSwaggerConfig => handleImplicitProperties
This should only be { noImplicitAdditionalProperties: "..." }, but only 'throw-on-extras' | 'silently-remove-extras' | 'ignore'

Reply:
I don't understand your comment here @WoH. Are you saying that the name here is wrong? Or that the type is wrong?

Either way, I still don't understand since it's a type that I use in a lot of different places and it represents a minimal interface for exactly the data that I'm expecting.

I'm also confused about the `'ignore'` in your comment about `'throw-on-extras' | 'silently-remove-extras' | 'ignore'` since I don't provide `'ignore'` as a valid option.

Reply:
I clarify below why I’d like to keep the data immutable.